sutherland_mike_ndvi.py

```diff
@@ -0,0 +1,603 @@
+    for dir in dirs:
+        loc_date = fetch_site_date(dir)
+        ndvi_calc = create_ndvi(dir, crop_extent, vals, valid_pixels)
+        ndvi_df_list.append(loc_date + ndvi_calc)
+
+# Final pandas df containing information for all Landsat scenes
+final_df = list_to_df(ndvi_df_list)
+final_df
+
+
+# In[9]:
+
+
+# Last sanity check before creating your plot (10 points)
+
+# Ensure that you call your dataframe at the bottom of the cell above
+# and that it has columns called: mean_ndvi and site
+
+# Ensure the data is stored in a dataframe.
+student_ndvi_df = _
+
+df_points = 0
+
+if isinstance(student_ndvi_df, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    df_points += 2
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Check that dataframe contains the appropriate number of NAN values
+if student_ndvi_df.mean_ndvi.isna().sum() == 15:
+    print('\u2705 Correct number of masked data values!')
+    df_points += 2
+else:
+    print('\u274C The amount of null data in your dataframe is incorrect.')
+
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_df.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    df_points += 3
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_df.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    df_points += 3
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Output for timer, # DO NOT MODIFY
+end_time = datetime.now()
+total_time = end_time - start_time
+print(
+    "Your total run time for processing the data was {0}.".format(total_time))
+
+print("\n \u27A1 You received {} out of 10 points for creating a dataframe.".format(
+    df_points))
+
+df_points
+
+
+# ## Task 2 Processing Explanation
+# 
+# Task 2 builds upon the overall workflow defined during Task 1 processing, with the only difference being it is scaled to process multiple Landsat scenes. Although I wanted to try to avoid using nested loops, doing so enabled me to only open the cropping extent once during each "site level" iteration ("outer loop"), which is somewhat more efficient than what I initially planned to do - simply defining a list at containing all the scenes and iterating through it, which would have forced me to open the crop extent during every iteration.
+
+# In[10]:
+
+
+# Plotting initialization
+f, ax = plt.subplots(figsize=(12, 8))
+
+# Drop nan values in df in order to correctly "connect the dots" in the plot
+final_df.dropna(inplace=True)
+
+# Plotting the data according to each respective site location
+for title, group in final_df.groupby("site"):
```

Comment:
I like the loop you used for your plot. I don't think I thought to do this in my code.